src/styles/main.scss

```diff
@@ -1,3 +1,146 @@
-body {
+$day-size: 100px;
+$border-width: 1px;
+$padding: 10px;
+$days-in-week: 7;
+$gap: 1px;
+$days-start: (
+  mon,
+  tue,
+  wed,
+  thu,
+  fri,
+  sat,
+  sun
+);
+
+@function find-index($list, $value) {
+  @return index($list, $value);
+}
+
+body,
+html {
+  height: 100%;
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar-container {
   margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
 }
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: $padding;
+  width: calc(7 * $day-size + 6 * $border-width + 2 * $padding);
+  position: relative;
+
+  .calendar__day {
+    width: $day-size;
+    height: $day-size;
+    background-color: #eee;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+    position: relative;
+    transition: all 0.5s;
+
+    &::before {
+      content: '';
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+
+    &::after {
+      content: '';
+      position: absolute;
+      top: 0;
+      left: 0;
+      width: 100%;
+      height: 100%;
+      border: $border-width solid #000;
+      box-sizing: border-box;
+    }
+  }
+
+    @for $i from 28 through 31 {
+      &.calendar-month-length--#{$i} {
+        .calendar__day:nth-child(n+#{$i + 1}) {
+          display: none;
+        }
+      }
+    }
+
+    @for $i from 1 through 31 {
+      .calendar__day:nth-child(#{$i})::before {
+        content: '#{$i}';
+      }
+    }
+
+    @each $day in $days-start {
+      &.start-day--#{$day} {
+        $start-index: find-index($days-start, $day);
+        $margin: ($start-index - 1);
+
+      .calendar__day {
+        &:nth-child(n + #{$margin + 1}):nth-child(-n + #{$margin + $days-in-week + 1}) {
+          margin-left: calc(#{$margin * ($day-size + 2 * $border-width)} + #{$padding});
+        }
+      }
+    }
+    }
+
+    &.calendar-start-day--sun {
+      .calendar__day:nth-child(1) {
+        margin-left: calc(6 * ($day-size + $border-width));
+      }
+    }
+
+    &.calendar-start-day--sat {
+      .calendar__day:nth-child(1) {
+        margin-left: calc(5 * ($day-size + $border-width));
+      }
+    }
+
+    &.calendar-start-day--fri {
+      .calendar__day:nth-child(1) {
+        margin-left: calc(4 * ($day-size + $border-width));
+      }
+    }
+
+    &.calendar-start-day--thu {
+      .calendar__day:nth-child(1) {
+        margin-left: calc(3 * ($day-size + $border-width));
+      }
```

Comment:
why do you need those styles if you have this `each` logic that creates all that's selectors